### ISSUE TITLE
Add custom HTTP headers on the request- and client-level

### ISSUE THREAD

**Issue by olivere** [olivere/elastic#1200]
Users should be able to specify custom HTTP headers both per request (at the service-level) and by passing a list of HTTP headers to sent along with every `PerformRequest` call.

**Comment by olivere** [comment]
See 3e93e182e21a89377ee466228064563aefde9680 for a description of the changes:

```
Add HTTP headers to all services and to client

This commit enables users to pass additional HTTP headers on every
service.

Users can use the `Header(name, value string)` and the
`Headers(http.Header)` methods to specify custom HTTP headers on the
request level.

Furthermore, users can specify a set of HTTP headers to
sent on every invocation of `PerformRequest` by using
`elastic.SetHeaders(http.Header)` when creating the client.

Request-level HTTP headers have preference over client-level HTTP
headers.

Close #1200
```

**Comment by olivere** [comment]
Backported to be included in 6.2.24 in 08e11cb.
